### ISSUE TITLE
Please investigate this version of Joe Kick .gltf in Sunrize vs Castle Game Engine

### ISSUE THREAD

**Issue by coderextreme** [create3000/sunrize#10]
[JoeSkin.zip](https://github.com/user-attachments/files/17815480/JoeSkin.zip)

**Comment by coderextreme** [comment]
Joe's upper torso is smeared: 
![image](https://github.com/user-attachments/assets/48276df9-3b73-48e2-9f66-fce8f6526d2a)

**Comment by create3000** [comment]
It looks like joint named 'Joe_vl5' is not referenced in the skeleton and on the wrong place, it is only a sibling of the humanoid.

**Comment by coderextreme** [comment]
I guess this is suffering from too many humanoids.   Sorry, I should have
looked at it closer,  when it worked in CGE with no errors, I got
overconfident.

Search for vl5_0 for example.

**Comment by create3000** [comment]
After inspecting the file again, there are not so many differences to CGE. The animation, when started, runs exactly the same as in CGE. This means that the initial values of the humanoid do not match the starting values of the animation, as the figure looks smashed when loaded. However, you can and must set this in Blender.

**Comment by coderextreme** [comment]
Agreed that the animation does not start in the same place as the humanoid
position. What I have been trying to do is set the centers according to the
local coordinate system.  What I can try to do next is insert a neutral 0
frame for each interpolator, or start animation at frame 1 instead of frame
0.

**Comment by create3000** [comment]
I found the culprit and with a few changes the scene-graph now looks like this:

<img width="1470" alt="Bildschirmfoto 2024-12-18 um 00 17 21" src="https://github.com/user-attachments/assets/c7e2fd75-779f-49be-ae76-42a125744c0a" />

**Comment by coderextreme** [comment]
Good news,  Joe Skin .gltf can load well in Sunrize with some tweaks,
according to Holger.  Maybe we can provide feed back to the blender gltf
export folks.  This Joe already looks good in castle viewer with my
testing.  I’m eager to try any updates to Sunrize.

Have a good night, Holger, good work!

I’ll probably be working on HAnim export in Blender again after confirming
import is working for Joe Skin.  I need to look carefully at the hierarchy
in Blender, then figure put the minimum I need to export to X3D.

Thanks!

John

**Comment by coderextreme** [comment]
The things I saw in the joekick export from Blender you sent was that it was duplicating the skeleton joints defs for transform without center then a use to define the center value. So the stock blender transform must not have a center parameter so it has to build some silly empty transforms to give a center value. Out of the dark ages before most all decided that a center value was part of a standard Transform.
Then, all 'accessory' geometry like the ball and circles of the stage and axis geometries got included in the 'skin' bindings instead of just the actual 'skin' field which is another reason it doesn't quite work right. Process the input so blender does not treat all geometry in the file as skin.
Seriously, even though the joekick is about the most simple example this looks like a big step with just a few failures, Nice Work. The best idea would be to fix blender so the Joint center is treated correctly(?)
Thanks and All Best,
Joe
-----Original Message-----

**Comment by coderextreme** [comment]
Probably the exporter has not set a Transform’s center for Blender empty
objects (from the object’s location) , or I create a transform instead of a
group.  But I’m probably confusing the importer with the loader.

Yes to extra fields on USE nodes, not covered yet.

Yes, my exporter is in flux, I’ve been focusing on import.

John

**Comment by coderextreme** [comment]
.glb seems to be broken!

John
[JoeSkin.zip](https://github.com/user-attachments/files/18214962/JoeSkin.zip)

**Comment by create3000** [comment]
Sorry, but I haven't published a new version yet, I won't do that until tomorrow, so be patient. The picture is a preview of the new version.

**Comment by create3000** [comment]
Test your [JoeSkin.zip](https://github.com/user-attachments/files/18214962/JoeSkin.zip) file and it will work then.

**Comment by create3000** [comment]
New version `1.7.38` of Sunrize with patch applied is now online and can be tested.
